File: src/main/java/springboottemplate/data_services/group/model/Group.java
package springboottemplate.data_services.group.model;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.*;
import springboottemplate.data_services.group.model.enums.HowToJoin;
import springboottemplate.data_services.group.model.enums.WhoCanCreatePosts;
import springboottemplate.data_services.group.model.enums.WhoCanSeePosts;

import java.util.ArrayList;
import java.util.List;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Builder

@Entity
public class Group {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    private String groupName;
    private String groupDescription;
    private byte[] groupImage;

    private WhoCanCreatePosts whoCanCreatePosts;
    private WhoCanSeePosts whoCanSeePosts;
    private HowToJoin howToJoin;

    @OneToMany
    @JsonManagedReference
    private List<GroupUser> groupUsers;

    @OneToMany
    @JsonManagedReference
    private List<GroupPost> groupPosts;

    @OneToMany
    private List<GroupUser> bannedUsers;

    public List<GroupUser> getGroupUsers() {
        if (groupUsers == null) groupUsers = new ArrayList<>();
        return groupUsers;
    }

    public List<GroupUser> getBannedUsers() {
        if (bannedUsers == null) bannedUsers = new ArrayList<>();
        return bannedUsers;
    }

    public List<GroupPost> getGroupPosts() {
        if (groupPosts == null) groupPosts = new ArrayList<>();
        return groupPosts;
    }
}

File: ui/src/services/group-service/model/Group.ts
import type {GroupUser} from "@/services/group-service/model/GroupUser";
import type {WhoCanCreatePosts} from "@/services/group-service/model/enums/WhoCanCreatePosts";
import type {WhoCanSeePosts} from "@/services/group-service/model/enums/WhoCanSeePosts";
import type {HowToJoin} from "@/services/group-service/model/enums/HowToJoin";
import type {GroupPost} from "@/services/group-service/model/GroupPost";

export interface Group {
    id: string;
    groupName: string;
    groupDescription: string;
    groupImage: string;
    whoCanCreatePosts: WhoCanCreatePosts;
    whoCanSeePosts: WhoCanSeePosts;
    howToJoin: HowToJoin;
    groupUsers: GroupUser[];
    groupPosts: GroupPost[];
    bannedUsers: GroupUser[];
}